lisa/tools/hugepages.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re
+from typing import Any, Set
+
+from lisa.executable import Tool
+from lisa.tools.echo import Echo
+from lisa.tools.free import Free
+from lisa.tools.ls import Ls
+from lisa.tools.lscpu import Lscpu
+from lisa.tools.mkfs import FileSystem
+from lisa.tools.mount import Mount
+from lisa.util import LisaException, SkippedException
+
+PATTERN_HUGEPAGE = re.compile(
+    r"^.*hugepages-(?P<hugepage_size_in_kb>\d+)kB.*",
+)
+
+
+class Hugepages(Tool):
+    @property
+    def command(self) -> str:
+        return ""
+
+    @property
+    def can_install(self) -> bool:
+        return False
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        self._hugepage_sizes: Set[str] = set()
+        self._hugepages_dir_path = "/sys/devices/system/node/node*/hugepages"
+
+    def get_hugepage_sizes_in_kb(self) -> Set[str]:
+        """
+        This function get the hugepage sizes in kB available on the system.
+        This is done by listing the hugepages directory
+        and extracting the size from the directory name.
+        e.g. [
+        "'/sys/devices/system/node/node0/hugepages/hugepages-16777216kB/'",
+        "'/sys/devices/system/node/node0/hugepages/hugepages-2048kB/'",
+        "'/sys/devices/system/node/node0/hugepages/hugepages-524288kB/'"
+        ]
+        """
+        ls = self.node.tools[Ls]
+        # if not ls.path_exists(self._hugepages_dir_path, sudo=True):
+        #     raise SkippedException(f"path: {self._hugepages_dir_path} does not exist")
+        hugepage_size_dir_names = ls.list_dir(self._hugepages_dir_path, sudo=True)
+        for hugepage_size_dir_name in hugepage_size_dir_names:
+            matched_hugepage = PATTERN_HUGEPAGE.match(hugepage_size_dir_name)
+            if not matched_hugepage:
+                raise LisaException(
+                    f"no hugepage size found in {hugepage_size_dir_name}"
+                )
+            self._hugepage_sizes.add(matched_hugepage.group("hugepage_size_in_kb"))
+        return self._hugepage_sizes
+
+    def _enable_hugepages(self, enable_gibibyte_hugepages: bool = False) -> None:
+        echo = self.node.tools[Echo]
+
+        meminfo = self.node.tools[Free]
+        nics_count = len(self.node.nics.get_nic_names())
+
+        numa_nodes = self.node.tools[Lscpu].get_numa_node_count()
+        request_pages_2mb = (nics_count - 1) * 1024 * numa_nodes
```

Comment:
As long as we're rewriting this; we should rip out this 2MiB or 1GiB assumption. ARM64 apparently has multiple other hugepage options, so we need to change how we enable them.

Consider how you would rewrite these `_enable_hugepages`  and `init_hugepages` functions now that we know there are more than 2 possible sizes of hugepages. We still want to check that we have enough memory; there's probably an easier way to do that. I started as a C dev and I have noticed I tend to write C even when I'm writing other languages 😅 Bad habit.

A good first step would be getting rid of the `enable_gibibyte_hugepages` boolean. That option is there because of the assumption that we're either using 2MiB or 1GiB hugepages. So now we can just rip it out. 

We don't want to assume the hugepage sizes; but, we *can* assume that **callers** will know what size they want. Tests will be testing a single size at a time; so it's good to assume that people will build tests where there is lots of shared code except one part where they call like "`run_test( .... , hugepage_size=(some constant) )`" . You could try something like creating an Enum of different sizes so the caller can check like...
```
supported_sizes = hugepages.get_hugepage_sizes() 
if not hugepages.HUGEPAGE_2MB in supported_sizes:
   raise SkippedException("2MiB hugepages not supported on this platform")
```
I'm sure @squirrelsc has thoughts, so let's check if this sounds good with him so we don't make you rewrite it twice 😜